src/main/java/org/newrelic/nrjmx/v2/JMXFetcher.java

```diff
@@ -386,10 +393,12 @@ private void getMBeanAttributes(ObjectName objectName, List<String> attributes,
             if (internalStat != null) {
                 internalStat.setSuccessful(true);
             }
-        } catch (ConnectException ce) {
+        } catch (JMXConnectionError je) {
+            throw je;
+        } catch (IOException io) {
```

Comment:
from documentation:
Throws:
IOException – A communication problem occurred when talking to the MBean server.